torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -2026,6 +2033,27 @@ void ONNXShapeTypeInference(
   }
 
   SpecialPostProcess(n);
+  // Get data propagation result from ONNX shape inference
+  for (auto output: n->outputs()) {
+    if (generated_shape_data.count(outputs_map[output->debugName()]) > 0) {
+      auto shape_data = generated_shape_data.find(outputs_map[output->debugName()])->second;
+      std::vector<::c10::ShapeSymbol> final_shape;
+      int rank = shape_data.dim_size();
+      final_shape.reserve(rank);
```

Comment:
You can avoid calling `reserve` by passing the size into the constructor, and then you'd do:
```c++
final_shape[i] = ONNXDimToShapeSymbol(shape_data.dim(i), symbol_map);
```